File: ojuz/ioi/2015/Day2/towns/grader.cpp
#include "towns.h"
#include <assert.h>
#include <stdio.h>
#include <stdlib.h>
#include "graderlib.c"

int main() {
    FILE *f;
    f = freopen("towns.in","r",stdin);
    assert(f != NULL);
    f = freopen("towns.out","w",stdout);
    assert(f != NULL);
    int ncase, R, N;
    int subtask;
    int ret;
    ret = scanf("%d%d",&subtask,&ncase);
    assert(ret == 2);
    for (int i = 0; i < ncase; i++) {
        ret = scanf("%d",&N);
	assert(ret == 1);
        _ini_query(N,subtask);
        R=hubDistance(N,subtask);
        printf("%d\n",R);
    }
    return 0;
}

File: ojuz/ioi/2015/Day2/towns/grader.pas
program grader;

uses towns, graderlib;

var
	_buffer: array [0..1023] of byte;
	_charsNumber: longint;
	_currentChar: longint;
	ncase, N, R : longint;
	subtask: longint;

begin
    assign(input, 'towns.in');
    reset(input);
    assign(output, 'towns.out');
    rewrite(output);

    _charsNumber := 0;
    _currentChar := 0;
    read(subtask, ncase);
    while ncase > 0 do
    begin
        dec(ncase);
        read(N);
        ini_query(N, subtask);
        R := hubDistance(N, subtask);
    writeln(r);
    end;
    close(input);
    close(output);
end.
